crypto.go

```diff
@@ -206,11 +206,11 @@ func blindBaseElement(blindingFactor btcec.ModNScalar) *btcec.PublicKey {
 type sharedSecretGenerator interface {
 	// generateSharedSecret given a public key, generates a shared secret
 	// using private data of the underlying sharedSecretGenerator.
-	generateSharedSecret(dhKey *btcec.PublicKey) (Hash256, error)
+	GenerateSharedSecret(dhKey *btcec.PublicKey) (Hash256, error)
 }
 
 // generateSharedSecret generates the shared secret by given ephemeral key.
-func (r *Router) generateSharedSecret(dhKey *btcec.PublicKey) (Hash256, error) {
+func (r *Router) GenerateSharedSecret(dhKey *btcec.PublicKey) (Hash256, error) {
```

Comment:
Godoc above not updated. 

Is this just so the implemented interface only has public methods, or that this method is to be used elsewhere?

Reply:
Godoc updated.

The method is used in #7139. Access is needed to the shared secret so that the type of onion encryptor can be 'upgraded' after reading the error structure parameters from the onion tlv.